src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/AuditService.kt

```diff
@@ -264,13 +265,23 @@ class AuditService(
       .transformToModelAuditEvents()
   }
 
-  private fun createAuditEvent(licence: Licence, currentUser: CommunityOffenderManager, summary: String, changes: Map<String, Any>) =
-    AuditEvent(
+  private fun createAuditEvent(
+    licence: Licence,
+    summary: String,
+    changes: Map<String, Any>,
+  ): AuditEvent {
+    val authUsername = SecurityContextHolder.getContext().authentication.name
+    val currentUser = communityOffenderManagerRepository.findByUsernameIgnoreCase(authUsername)
+
+    return AuditEvent(
       licenceId = licence.id,
-      username = currentUser.username,
-      fullName = "${currentUser.firstName} ${currentUser.lastName}",
-      summary = "$summary for ${licence.forename} ${licence.surname}",
+      username = currentUser?.username ?: "SYSTEM",
+      fullName = if (currentUser != null) "${currentUser.firstName} ${currentUser.lastName}" else "SYSTEM",
+      eventType = if (currentUser == null) AuditEventType.SYSTEM_EVENT else AuditEventType.USER_EVENT,
+      summary =
+      if (currentUser != null) "$summary for ${licence.forename} ${licence.surname}" else "Unable to find offender for $authUsername",
```

Comment:
don't think we need the warning message here - its more important to have the summary message so could leave this as is?